test/TuDa.CIMS.Api.Test/Controllers/AssetItemControllerTest.cs

```diff
@@ -216,4 +219,113 @@ public async Task GetPaginatedAsync_ShouldReturnCorrectPagination_WhenPagination
+            consumableFaker,
+            new SolventFaker(),
+            new GasCylinderFaker(),
+        ];
+
+        const string updatedName = "Updated";
+        const int updatedAmount = 15;
+
+        List<UpdateAssetItemDto> updateAssetItemDtos =
+        [
+            new UpdateChemicalDto(),
+            new UpdateConsumableDto
+            {
+                Name = updatedName,
+                StockUpdate = new StockUpdateDto(updatedAmount, TransactionReasons.Restock)
+            },
+            new UpdateSolventDto(),
+            new UpdateGasCylinderDto(),
+        ];
+
+        await _dbContext.AssetItems.AddRangeAsync(assetItems);
+        await _dbContext.SaveChangesAsync();
+
+        var previousAmount = consumableFaker.Amount;
+        foreach (var (assetItem, updateAssetItemDto) in assetItems.Zip(updateAssetItemDtos))
+        {
+            var updatedItem = assetItem with { Name = updatedName };
+            updateAssetItemDto.Name = updatedName;
+
+            // Act
+            var response = await _client.PatchAsync(
+                $"api/asset-items/{assetItem.Id}",
+                JsonContent.Create(updateAssetItemDto)
+            );
+
+            // Assert
+            response.IsSuccessStatusCode.Should().BeTrue();
+
+            var result = await _dbContext.AssetItems.SingleAsync(i => i.Id == assetItem.Id);
+            result
+                .Should()
+                .BeEquivalentTo(updatedItem, options => options.Excluding(item => item.UpdatedAt));
+            result.UpdatedAt.Should().NotBeNull();
+
+            if (assetItem is Consumable)
+            {
+                var transaction = await _dbContext.ConsumableTransactions
+                    .Where(t => t.Consumable.Id == assetItem.Id)
+                    .OrderByDescending(t => t.Date)
+                    .FirstOrDefaultAsync();
+
+                transaction.Should().NotBeNull();
+                transaction!.AmountChange.Should().Be(updatedAmount - previousAmount);
+                transaction.TransactionReason.Should().Be(TransactionReasons.Restock);
+            }
+        }
+    }
+   /* [Fact]
+    public async Task CreateAsync_ShouldCreateConsumableTransaction_WhenConsumableIsCreated()
+    {
+        // Arrange
+        var room = new Room { Name = "test", Id = Guid.NewGuid() };
+        await _dbContext.Rooms.AddAsync(room);
+        await _dbContext.SaveChangesAsync();
+
+        var createConsumable = new CreateConsumableDto
+        {
+            ItemNumber = "12345",
+            Name = "Test Consumable",
+            Note = "Test Note",
+            Price = 50,
+            Shop = "Test Shop",
+            RoomId = room.Id,
+            Amount = 10,
+            Manufacturer = "Test Manufacturer",
+            SerialNumber = "ABC123"
+        };
+
+        // Act
+        var response = await _client.PostAsync(
+            "api/asset-items",
+            JsonContent.Create(createConsumable)
+        );
```

Comment:
🔧 Specify that this should be serialized as `AssetItemCreateConsumable`
```suggestion
        var response = await _client.PostAsync(
            "api/asset-items",
            JsonContent.Create<CreateAssetItemDto>(createConsumable)
        );
```